src/transformers/pipelines/object_detection.py

```diff
@@ -0,0 +1,177 @@
+
+        Return:
+            A list of dictionaries or a list of list of dictionaries containing the result. If the input is a single
+            image, will return a list of dictionaries, if the input is a list of several images, will return a list of
+            list of dictionaries corresponding to each image.
+
+            The dictionaries contain the following keys:
+
+            - **label** (:obj:`str`) -- The class label identified by the model.
+            - **score** (:obj:`float`) -- The score attributed by the model for that label.
+            - **box** (:obj:`List[Dict[str, int]]`) -- The bounding box of detected object in image's original size.
+        """
+        is_batched = isinstance(images, list)
+
+        if not is_batched:
+            images = [images]
+
+        images = [self.load_image(image) for image in images]
+
+        with torch.no_grad():
+            inputs = self.feature_extractor(images=images, return_tensors="pt")
+            outputs = self.model(**inputs)
+
+            if self.framework == "pt":
+                target_sizes = torch.IntTensor([[im.height, im.width] for im in images])
+            else:
+                raise ValueError("The ObjectDetectionPipeline is only available in PyTorch.")
+
+            raw_annotations = self.feature_extractor.post_process(outputs, target_sizes)
+            annotations = []
+            for annotation in raw_annotations:
+                keep = annotation["scores"] > threshold
+                scores = annotation["scores"][keep]
+                labels = annotation["labels"][keep]
+                boxes = annotation["boxes"][keep]
+
+                annotation["scores"] = scores.tolist()
+                annotation["labels"] = [self.model.config.id2label[label.item()] for label in labels]
+                annotation["boxes"] = [self._get_bounding_box(box) for box in boxes]
+
+                # {"scores": [...], ...} --> [{"score":x, ...}, ...]
+                keys = ["score", "label", "box"]
+                annotation = [
+                    dict(zip(keys, vals))
+                    for vals in zip(annotation["scores"], annotation["labels"], annotation["boxes"])
+                ]
+
+                annotations.append(annotation)
+
+        if not is_batched:
+            return annotations[0]
+
+        return annotations
+
+    def _get_bounding_box(self, box: "torch.Tensor") -> Dict[str, int]:
+        """
+        Turns list [xmin, xmax, ymin, ymax] into dict { "xmin": xmin, ... }
+
+        Args:
+            box (torch.Tensor): Tensor containing the coordinates in corners format.
+
+        Returns:
+            bbox (Dict[str, int]): Dict containing the coordinates in corners format.
+        """
+        if self.framework != "pt":
+            # guarding `box: "torch.Tensor"`
+            raise ValueError("The ObjectDetectionPipeline is only available in PyTorch.")
+        xmin, ymin, xmax, ymax = box.int().tolist()
```

Comment:
@Narsil guarding "torch.Tensor" here. Please let me know what you think about it